programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -90,12 +90,47 @@ pub mod gateway {
         Ok(())
     }
 
-    // whitelisting SPL tokens
-    pub fn whitelist_spl_mint(_ctx: Context<Whitelist>) -> Result<()> {
+    pub fn whitelist_spl_mint(
```

Comment:
Might be I,m not familiar yet with Solana enought but where is the state transition/side effect in this function?